float32 X_dir
float32 Y_dir
float32 Z_dir
float32 MaxVelPercentage
int32   Mode
